mongo/client.go

```diff
@@ -288,31 +288,27 @@ func (c *Client) endSessions(ctx context.Context) {
 		return
 	}
 
-	ids := c.sessionPool.IDSlice()
```

Comment:
I realized we were doing extra work here because we make a big BSON array with all the IDs but continuously overwrite it with smaller arrays created below so I removed this code and changed the `NewEndSessions` call to pass `nil` instead.